### DISCUSSION TITLE
Ability to change wildcard permission dot (.) to colon (:)

### DISCUSSION THREAD

**Discussion by doraemonxxx** [spatie/laravel-permission#2110] [Q&A] [answered]
Hello there, is there any way to change the dot (.) into a colon (:) when using wildcard permission?

**for example**
user has the permission of `admin:*`

user can access all related admin permission even like 
`admin:list`
`admin:create`
`admin:save`

even though users don't have these permissions. I would like to use the colon because some  requirements in my task.

Thank you!
Cheers !🍻

**Comment by doraemonxxx** [comment]
after digging some files and code, the delimiter variable can be found in **WildcardPermission.php** but I don't know how to change `PART_DELIMITER` into a colon (:)

**Comment by erikn69** [comment, marked as answer]
you could overwrite `hasWildcardPermission` with your custom `WildcardPermission` class
https://github.com/spatie/laravel-permission/blob/96466b07c9a416fcc095a9097e87131e082e60af/src/Traits/HasPermissions.php#L163-L192

**Reply by doraemonxxx** [reply]
so I will create a new model that extends spatie model and then overwrite the `hasWildcardPermission ` method?

**Reply by erikn69** [reply]
No, you did write
>user has the permission of admin:*

Only overwrite on user

**Reply by doraemonxxx** [reply]
> No, you did write
> 
> > user has the permission of admin:*
> 
> Only overwrite on user

I will use the `HasPermissions` trait in the User model and then overwrite the `hasWildcardPermission ` ? correct ?

for example: 
`use Spatie\Permission\Traits\HasRoles;
use Spatie\Permission\Traits\HasPermissions;

class User extends Authenticatable
{
    use  HasRoles, HasPermissions;
}

protected function hasWildcardPermission($permission, $guardName = null): bool{
 //overwrite the method.
}`


thanks!

**Reply by erikn69** [reply]
`hasRoles` already use `hasPermissions`
```php
class User extends Authenticatable {
    use HasRoles;

    protected function hasWildcardPermission($permission, $guardName = null): bool{
        //overwrite the method with the custom wilcard class
    }
}
```

**Comment by adesege** [comment]
You can now replace WildcardPermission class in permission.php config.

Add `permission.wildcard_permission` property to your config file. The value will be a reference to your new WildcardPermission class.

Then do:

```php
class WildcardPermission extends \Spatie\Permission\WildcardPermission
{
    public const PART_DELIMITER = ':';
}

``` 

Much cleaner than overriding `hasWildcardPermission`

**Reply by drbyte** [reply]
Thanks for the reminder. I've added this to the (upcoming v6, to be released soon) config file for easy reference.
